src/mongocxx/private/database.hpp

```diff
@@ -34,6 +34,19 @@ class database::impl {
         name(std::move(name))
     {}
 
+    impl(const impl& i) : 
+        database_t{libmongoc::database_copy(i.database_t)},
+        client_impl{i.client_impl},
+        name{i.name} 
+    {}
+
+    impl& operator=(const impl& i) {
+        database_t = libmongoc::database_copy(i.database_t);
```

Comment:
Right, but as you said, that's not using this operator.  If we were using the copy assignment operator, we would have to explicitly destroy the database_t.  So if we're going to implement this method, we should do that, even if we won't end up using the code now.